flybywiresim/aircraft
# feat(mcdu): subsystem requests and scratchpads, and annunciators

Fixes #[issue_no]

## Summary of Changes

This PR implements initial support for MCDU annunciators, and subsystem requests. The MCDU MENU page is largely re-implemented with correct behaviour now this is possible. Additionally scratchpads for each subsystem are split to correct behaviour (although they are not actually managed by the subsystem computers themselves in this PR).

## Screenshots (if necessary)


![image](https://github.com/flybywiresim/aircraft/assets/9995998/0836888e-64e7-49d5-bb46-55dd364bcec4)
![image](https://github.com/flybywiresim/aircraft/assets/9995998/a4d7889a-7a53-4f8d-9623-f38d0fb1123a)
![image](https://github.com/flybywiresim/aircraft/assets/9995998/df4df4b5-8046-4ffd-ada1-cb105dc35437)
![image](https://github.com/flybywiresim/aircraft/assets/9995998/8ad511a2-e1dd-4601-a2b5-35ae748f98cb)
![image](https://github.com/flybywiresim/aircraft/assets/9995998/2c76d9cc-3bf7-4ba9-9a96-49a490f79baf)

## References

ARINC739, FMGC/MCDU docs. and videos.




## Additional context



Discord username (if different from GitHub):

## Testing instructions

- Check that each subsystem scratchpad is independent.
- Trigger the MCDU MENU annunciator on the MCDU by requesting WX in the ATSU and then quickly swapping to the FMGC subsystem (e.g. press the F-PLN key).
- Check that the MCDU MENU screen functions properly, and that the scratchpad cannot be edited with MCDU keypad.
- Check that MCDU keyboard entry works.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
